app/code/Magento/Catalog/Test/Unit/ViewModel/Product/Checker/AddToCompareAvailabilityTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Catalog\Test\Unit\ViewModel\Product\Checker;
+
+use Magento\Catalog\ViewModel\Product\Checker\AddToCompareAvailability;
+use Magento\CatalogInventory\Api\StockConfigurationInterface;
+use Magento\Catalog\Model\Product;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use PHPUnit\Framework\MockObject\MockObject;
+
+/**
+ * Unit test for Magento\Catalog\ViewModel\Product\Checker\AddToCompareAvailability.
+ */
+class AddToCompareAvailabilityTest extends \PHPUnit\Framework\TestCase
+{
+
+    /**
+     * @var AddToCompareAvailability
+     */
+    private $viewModel;
+
+    /**
+     * @var StockConfigurationInterface|MockObject 
+     */
+    protected $stockConfiguration;
```

Comment:
Add suffix `Mock` to use `stockConfigurationMock` as a name, it's clear that we're facing Mock, then.

Reply:
Good point, I will update with suffix